src/protocols/screencopy.rs

```diff
@@ -16,12 +20,13 @@ use smithay::reexports::wayland_server::{
     Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New, Resource,
 };
 use smithay::utils::{Physical, Point, Rectangle, Size};
-use smithay::wayland::shm;
+use smithay::wayland::{dmabuf, shm};
 
-// We do not support copy_with_damage() semantics yet.
-const VERSION: u32 = 1;
+const VERSION: u32 = 3;
 
-pub struct ScreencopyManagerState;
+pub struct ScreencopyManagerState {
+    damage_trackers: HashMap<Output, ScreencopyDamageTracker>,
```

Comment:
There needs to be a damage tracker per screencopy manager instance, rather than per output. (It can stay per-output, but every output can have multiple independent damage trackers.)

This is because multiple screencopy clients can be connected at once, and they will all have separate damage accumulation.

The protocol says about damage:

> area that has changed since the last copy request that was derived from the current screencopy manager instance

Reply:
Also, I wonder what should happen if a client creates two screencopy frames at once and calls copy_with_damage() on both of them without waiting for each other? I guess in this case we can fail the second one? Or, if I'm understanding the protocol right, we can queue up the second one to be processed as if it was called right after we send the first one to the client.

Reply:
> Also, I wonder what should happen if a client creates two screencopy frames at once and calls copy_with_damage() on both of them without waiting for each other? 

If that's the same frame then that's already used, but if different I think it is currently serializing them?